tests/test_blurb_add.py

```diff
@@ -0,0 +1,218 @@
+import re
+
+import pytest
+
+from blurb import blurb
+
+
+ALLOWED_SECTION_IDS = list(map(str, range(1 + len(blurb.sections), 1)))
+
+
+def test_valid_no_issue_number():
+    assert blurb._extract_issue_number(None) is None
+    res = blurb._update_blurb_template(issue=None, section=None)
+    lines = res.splitlines()
+    assert f'.. gh-issue:' not in lines
+    assert f'.. gh-issue: ' in lines
+    for line in lines:
+        assert not line.startswith('.. section: ')
+
+
+@pytest.mark.parametrize(('issue', 'expect'), [
+    # issue given by their number
+    ('12345', '12345'),
+    ('12345 ', '12345'),
+    (' 12345', '12345'),
+    (' 12345 ', '12345'),
+    # issue given by their number and a 'gh-' prefix
+    ('gh-12345', '12345'),
+    ('gh-12345 ', '12345'),
+    (' gh-12345', '12345'),
+    (' gh-12345 ', '12345'),
+    # issue given by their URL (no protocol)
+    ('github.com/python/cpython/issues/12345', '12345'),
+    ('github.com/python/cpython/issues/12345 ', '12345'),
+    (' github.com/python/cpython/issues/12345', '12345'),
+    (' github.com/python/cpython/issues/12345 ', '12345'),
+    # issue given by their URL (with protocol)
+    ('https://github.com/python/cpython/issues/12345', '12345'),
+    ('https://github.com/python/cpython/issues/12345 ', '12345'),
+    (' https://github.com/python/cpython/issues/12345', '12345'),
+    (' https://github.com/python/cpython/issues/12345 ', '12345'),
+])
+def test_valid_issue_number(issue, expect):
+    actual = blurb._extract_issue_number(issue)
+    assert actual == expect
```

Comment:
`expect` is always "12345", we can remove it here:
```suggestion
@pytest.mark.parametrize('issue', [
    # issue given by their number
    '12345',
    '12345 ',
    ' 12345',
    ' 12345 ',
    # issue given by their number and a 'gh-' prefix
    'gh-12345',
    'gh-12345 ',
    ' gh-12345',
    ' gh-12345 ',
    # issue given by their URL (no protocol)
    'github.com/python/cpython/issues/12345',
    'github.com/python/cpython/issues/12345 ',
    ' github.com/python/cpython/issues/12345',
    ' github.com/python/cpython/issues/12345 ',
    # issue given by their URL (with protocol)
    'https://github.com/python/cpython/issues/12345',
    'https://github.com/python/cpython/issues/12345 ',
    ' https://github.com/python/cpython/issues/12345',
    ' https://github.com/python/cpython/issues/12345 ',
])
def test_valid_issue_number(issue):
    actual = blurb._extract_issue_number(issue)
    assert actual == "12345"

    res = blurb._update_blurb_template(issue=issue, section=None)

    lines = res.splitlines()
    assert '.. gh-issue:' not in lines
    for line in lines:
        assert not line.startswith('.. section: ')

    assert '.. gh-issue: 12345' in lines
    assert '.. gh-issue: ' not in lines
```

Reply:
I had in mind that we could maybe in the future extend the tests but actually, I don't think it's necessary so thank you!